.github/workflows/github-actions.yaml

```diff
@@ -0,0 +1,50 @@
+name: Docker build
+
+on:
+  push:
+    branches: ['main']
+
+env:
+  VERSION: 0.0.1
+  IMAGE_NAME: zjmorgan/subhkl:${{ version }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Download repo
+        uses: actions/checkout@v4
+
+      - name: Code ornl gov login
+        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
+        with:
+          registry: code.ornl.gov:4567
+          password: ${{ secrets.rse_mantid_docker_token }}
+          username: ${{ secrets.rse_mantid_docker_username }}
```

Comment:
Just to check, is this username and token tied to your credentials @SmithRWORNL?

Or a project based token?

Reply:
@marshallmcdonnell My username, with a project based token.

Reply:
Awesome :+1: 

We also have a service account we can swap to for other projects.

I'll send it via a direct communication.